File: code/community/1306330637468176395/deepgram-diarization-pricing.js
const axios = require('axios');
require('dotenv').config();

const apiKey = process.env.DEEPGRAM_API_KEY;
const audioUrl = process.env.AUDIO_URL;

async function transcribeAudio() {
    try {
        const response = await axios.post('https://api.deepgram.com/v1/listen', {
            url: audioUrl,
            diarize: true,
        }, {
            headers: {
                'Authorization': `Token ${apiKey}`,
                'Content-Type': 'application/json',
            },
        });

        console.log('Transcription result:', response.data);
    } catch (error) {
        console.error('Error:', error.response.status, error.response.data);
    }
}

transcribeAudio();

File: code/community/1306330637468176395/deepgram-diarization-pricing.rs
use std::env;
use reqwest::header::{AUTHORIZATION, CONTENT_TYPE};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let api_key = env::var("DEEPGRAM_API_KEY").expect("DEEPGRAM_API_KEY not set");
    let audio_url = env::var("AUDIO_URL").expect("AUDIO_URL not set");

    let client = reqwest::Client::new();
    let response = client.post("https://api.deepgram.com/v1/listen")
        .header(AUTHORIZATION, format!("Token {}", api_key))
        .header(CONTENT_TYPE, "application/json")
        .json(&serde_json::json!({
            "url": audio_url,
            "diarize": true,
        }))
        .send()
        .await?;

    let result: serde_json::Value = response.json().await?;
    println!("Transcription result: {:?}", result);

    Ok(())
}